ItemPool.py

```diff
@@ -813,368 +323,226 @@ def collect_heart_container(world, pool):
-    else:
-        skulltula_locations_final.extend([
-            'Spirit Temple GS Metal Fence',
-            'Spirit Temple GS Sun on Floor Room',
-            'Spirit Temple GS Hall After Sun Block Room',
-            'Spirit Temple GS Boulder Room',
-            'Spirit Temple GS Lobby'])
-    if world.dungeon_mq['Shadow Temple']:
-        skulltula_locations_final.extend([
-            'Shadow Temple MQ GS Falling Spikes Room',
-            'Shadow Temple MQ GS Wind Hint Room',
-            'Shadow Temple MQ GS After Wind',
-            'Shadow Temple MQ GS After Ship',
-            'Shadow Temple MQ GS Near Boss'])
-    else:
-        skulltula_locations_final.extend([
-            'Shadow Temple GS Like Like Room',
-            'Shadow Temple GS Falling Spikes Room',
-            'Shadow Temple GS Single Giant Pot',
-            'Shadow Temple GS Near Ship',
-            'Shadow Temple GS Triple Giant Pot'])
-    if world.dungeon_mq['Bottom of the Well']:
-        skulltula_locations_final.extend([
-            'Bottom of the Well MQ GS Basement',
-            'Bottom of the Well MQ GS Coffin Room',
-            'Bottom of the Well MQ GS West Inner Room'])
-    else:
-        skulltula_locations_final.extend([
-            'Bottom of the Well GS West Inner Room',
-            'Bottom of the Well GS East Inner Room',
-            'Bottom of the Well GS Like Like Cage'])
-    if world.dungeon_mq['Ice Cavern']:
-        skulltula_locations_final.extend([
-            'Ice Cavern MQ GS Scarecrow',
-            'Ice Cavern MQ GS Ice Block',
-            'Ice Cavern MQ GS Red Ice'])
-    else:
-        skulltula_locations_final.extend([
-            'Ice Cavern GS Spinning Scythe Room',
-            'Ice Cavern GS Heart Piece Room',
-            'Ice Cavern GS Push Block Room'])
-    if world.settings.tokensanity == 'off':
-        for location in skulltula_locations_final:
-            placed_items[location] = 'Gold Skulltula Token'
-    elif world.settings.tokensanity == 'dungeons':
-        for location in skulltula_locations_final:
-            if world.get_location(location).scene >= 0x0A:
-                placed_items[location] = 'Gold Skulltula Token'
+    # Use the vanilla items in the world's locations when appropriate.
+    for location in world.get_locations():
+        if location.vanilla_item is None:
+            continue
+
+        item = location.vanilla_item
+        shuffle_item = None  # None for don't handle, False for place item, True for add to pool.
+
+        # Always Placed Items
+        if location.vanilla_item in ['Zeldas Letter', 'Triforce', 'Scarecrow Song',
+                                     'Deliver Letter', 'Time Travel', 'Bombchu Drop'] \
+                or location.type == 'Drop':
+            shuffle_item = False
+
+        # Gold Skulltula Tokens
+        elif location.vanilla_item == 'Gold Skulltula Token':
+            shuffle_item = world.settings.tokensanity == 'all' \
```

Comment:
nit: wrap in parens instead of using `\\`

Reply:
Resolved.